internal/controller/replication.storage/status.go

```diff
@@ -46,26 +50,30 @@ func setPromotedCondition(conditions *[]metav1.Condition, observedGeneration int
 }
 
 // sets conditions when volume promotion was failed.
-func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64) {
+func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64, message, messageFromDriver string) {
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            message,
 		Type:               v1alpha1.ConditionCompleted,
 		Reason:             v1alpha1.FailedToPromote,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            fmt.Sprintf("%s: %s", v1alpha1.MessageFailedPromoted, messageFromDriver),
```

Comment:
It is little bit confusing - we have:
- message
- messageFromDriver

It seems that message is related to Completed condition, and messageFromDriver is related to DegradtedCondition. But the name do not give any hint about these relations.

Reply:
message is any error message that is being detected in csiaddons, and messagefromdriver is the rpc msgs that comes from ceph csi driver